src/Morphir/JsonSchema/Backend.elm

```diff
@@ -105,55 +105,96 @@ extractTypes modName definition =
+mapTypeDefinition (( path, name ) as qualifiedName) definition =
     case definition of
-        Type.TypeAliasDefinition typeArgs typ ->
-            mapType typ
-                |> Maybe.map (Tuple.pair typeName)
+        Type.TypeAliasDefinition _ typ ->
+            [ ( mapQualifiedName qualifiedName, mapType typ ) ]
+
+        Type.CustomTypeDefinition _ accessControlledCtors ->
+            let
+                oneOfs2 =
+                    accessControlledCtors.value
+                        |> Dict.toList
+                        |> List.map
+                            (\( ctorName, ctorArgs ) ->
+                                let
+                                    ctorNameString =
+                                        ctorName |> Name.toTitleCase
+                                in
+                                if List.isEmpty ctorArgs then
+                                    Const ctorNameString
+
+                                else
+                                    Array
+                                        (TupleType
+                                            (Const (ctorName |> Name.toTitleCase)
+                                                :: (ctorArgs
+                                                        |> List.map (Tuple.second >> mapType)
+                                                   )
+                                            )
+                                            ((ctorArgs |> List.length) + 1)
+                                        )
+                                        False
+                            )
+            in
+            [ ( (path |> Path.toString Name.toTitleCase ".") ++ "." ++ (name |> Name.toTitleCase), OneOf oneOfs2 ) ]
+
+
+mapType : Type ta -> SchemaType
+mapType typ =
+    case typ of
+        Type.Reference _ (( packageName, moduleName, localName ) as fQName) argTypes ->
+            case ( FQName.toString fQName, argTypes ) of
+                ( "Morphir.SDK:Basics:int", [] ) ->
+                    Integer
 
-        Type.CustomTypeDefinition typeArgs accessControlledConstructors ->
-            Nothing
+                ( "Morphir.SDK:Decimal:decimal", [] ) ->
+                    String
 
+                ( "Morphir.SDK:String:string", [] ) ->
+                    String
 
-mapType : Type ta -> Maybe SchemaType
-mapType typ =
-    case typ of
-        Type.Reference a fQName types ->
-            case FQName.toString fQName of
-                "Morphir.SDK:Basics:int" ->
-                    Just Integer
+                ( "Morphir.SDK:Char:char", [] ) ->
+                    String
 
-                "Morphir.SDK:String:string" ->
-                    Just String
+                ( "Morphir.SDK:Basics:float", [] ) ->
+                    Number
 
-                "Morphir.SDK:Basics:float" ->
-                    Just Number
+                ( "Morphir.SDK:Tuple:tuple", [] ) ->
+                    Number
```

Comment:
This is not right. A tuple definitely should not map to a number. Also, I don't think this type even exists in the SDK.